libs/JSystem/JStudio/JStudio/functionvalue.cpp

```diff
@@ -519,176 +294,292 @@ SECTION_SDATA2 static f64 lit_800 = 0.5;
 /* 80455430-80455438 003A30 0008+00 1/1 0/0 0/0 .sdata2          @801 */
 SECTION_SDATA2 static f64 lit_801 = 2.0 / 3.0;
 
-/* 80281774-802817D8 27C0B4 0064+00 1/1 0/0 0/0 .text
- * interpolateValue_BSpline_uniform__Q27JStudio13functionvalueFddddd */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void JStudio::functionvalue::interpolateValue_BSpline_uniform(f64 param_0, f64 param_1,
-                                                                  f64 param_2, f64 param_3,
-                                                                  f64 param_4) {
-    nofralloc
-#include "asm/JSystem/JStudio/JStudio/functionvalue/interpolateValue_BSpline_uniform__Q27JStudio13functionvalueFddddd.s"
+/* ############################################################################################## */
+/* 80455438-80455440 003A38 0008+00 3/3 0/0 0/0 .sdata2          @842 */
+SECTION_SDATA2 static f64 lit_842 = 2.0;
+
+/* ############################################################################################## */
+/* 80455440-80455448 003A40 0008+00 1/1 0/0 0/0 .sdata2          @867 */
+SECTION_SDATA2 static f64 lit_867 = -1.0;
+
+/* ############################################################################################## */
+/* 804511E8-804511F0 0006E8 0001+07 1/1 0/0 0/0 .sbss            @569 */
+static u8 lit_569[1 + 7 /* padding */];
+
+/* ############################################################################################## */
+/* 80455448-80455450 003A48 0008+00 1/1 0/0 0/0 .sdata2          @1517 */
+SECTION_SDATA2 static f64 lit_1517 = 4503599627370496.0 /* cast u32 to float */;
+
+namespace JStudio {
+
+/* ############################################################################################## */
+/* 8039A9F0-8039AA00 -00001 0010+00 1/1 0/0 0/0 .rodata
+ * gapfnExtrapolateParameter___Q27JStudio27@unnamed@functionvalue_cpp@ */
+const ExtrapolateParameter gapfnExtrapolateParameter_[4] = {
+    (ExtrapolateParameter)extrapolateParameter_raw__Q27JStudio13functionvalueFdd,
+    (ExtrapolateParameter)extrapolateParameter_repeat__Q27JStudio13functionvalueFdd,
+    (ExtrapolateParameter)extrapolateParameter_turn__Q27JStudio13functionvalueFdd,
+    (ExtrapolateParameter)extrapolateParameter_clamp__Q27JStudio13functionvalueFdd,
```

Comment:
```suggestion
    (ExtrapolateParameter)JStudio::functionvalue::extrapolateParameter_raw,
    (ExtrapolateParameter)JStudio::functionvalue::extrapolateParameter_repeat,
    (ExtrapolateParameter)JStudio::functionvalue::extrapolateParameter_turn,
    (ExtrapolateParameter)JStudio::functionvalue::extrapolateParameter_clamp,
```
Does something like this work? Maybe that makes it possible to get rid of the casts